contentcuration/contentcuration/static/js/edit_channel/image/views.js

```diff
@@ -41,27 +44,35 @@ var ThumbnailUploadView = BaseViews.BaseView.extend({
     events: {
         'click .remove_image ' : 'remove_image',
         'click .open_thumbnail_generator': 'open_thumbnail_generator',
-        'click .get_image': 'get_image'
+        'click .crop_image': 'create_croppie',
+        'click .cancel_image': 'cancel_croppie',
+        'click .submit_image': 'submit_croppie'
     },
     render: function() {
         if(this.allow_edit){
             this.$el.html(this.template({
                 picture : this.get_thumbnail_url(),
                 selector: this.get_selector(),
-                show_generate: this.model.get('kind') != undefined
+                show_generate: this.model.get('kind') != undefined,
+                show_crop: true
             }));
             _.defer(this.create_dropzone, 1);
-            _.defer(this.create_croppie, 2);
         }else{
             this.$el.html(this.preview_template({
                 picture : this.get_thumbnail_url(),
                 name: this.model.get('title')
             }));
         }
     },
-    get_thumbnail_url:function(){
+    get_thumbnail_url:function(ignore_encoding){
         var thumbnail = _.find(this.model.get('files'), function(f){ return f.preset.thumbnail; });
-        if(this.image_url){ return this.image_url; }
+        if(!ignore_encoding && this.thumbnail_encoding){
+            if(typeof this.thumbnail_encoding === "string"){
```

Comment:
Also, do we really need a separate model field? I know browsers are smart enough to read `data` URIs or HTTP URLs: https://css-tricks.com/data-uris/

so I think we can just throw either a base64 or HTTP URL to a browser, and it will do the right thing.